starter-code/main.js

```diff
@@ -0,0 +1,376 @@
+                {
+                  count++;
+
+                }
+
+              }
+
+              console.log(count);
+
+              rl.question(`\nType enter to continue: `, (answer) =>{ mainMenu()} );
+            }
+          });
+            break;
+          case "13":
+          db.collection('companies').find({"relationships.person.permalink": "david-ebersman"}).toArray((error, result) => {
+            if (error)
+            {
+              console.log(error);
+              rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+            }
+            else
+            {
+              // console.log(result);
+              for(var i = 0 ; i  <result.length; i++)
+              {
+                console.log(result[i].name);
+              }
+
+              rl.question(`\nType enter to continue: `, (answer) =>{ mainMenu()} );
+            }
+          });
+            break;
+          case "14":
+          db.collection('companies').find({"name": "Facebook"}).toArray((error, result) => {
+            if (error)
+            {
+              console.log(error);
+              rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+            }
+            else
+            {
+              for(var i = 0 ; i  <result[0].competitions.length; i++)
+              {
+                console.log(result[0].competitions[i].competitor.name);
+              }
+
+              rl.question(`\nType enter to continue: `, (answer) =>{ mainMenu()} );
+            }
+          });
+            break;
+          case "15":
+          db.collection('companies').find({tag_list:{$regex: 'social-networking'}}).toArray((error, result) => {
+            if (error)
+            {
+              console.log(error);
+              rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+            }
+            else
+            {
+              for(var i = 0 ; i  <result.length; i++)
+              {
+                console.log(result[i].name);
+                console.log(result[i].tag_list);
+                console.log("------");
+
+              }
+
+
+              rl.question(`\nType enter to continue: `, (answer) =>{ mainMenu()} );
+            }
+          });
+            break;
+          case "16":
+
+          db.collection('companies').find({ $and: [{tag_list:{$regex: 'social-networking'}},{founded_year: { $gt: 2002, $lte: 2016 }}]}).count((error, result) => {
+            if (error)
+            {
+              console.log(error);
+              rl.question(`\nType enter to continue: `, (answer) => { mainMenu() });
+            }
+            else
+            {
+                console.log(result);
+
+              rl.question(`\nType enter to continue: `, (answer) =>{ mainMenu()} );
+            }
+          });
+            break;
+          case "17":
+          db.collection('companies').find({"offices.city": "London"}).toArray((error, result) => {
```

Comment:
next time, because you just need name and offices try to use projections to return less data from the db